### PR TITLE
WIP: Add python-toolchain

### PR DESCRIPTION
**Pull request by jakirkham**
Partially resolves: https://github.com/conda-forge/staged-recipes/issues/528

This is a "toolchain" for configuring our Python installs. This PR includes a few pieces that still need a bit more polishing.

As a note, this package is designed intentionally to not overlap with the `toolchain` package. The reason is the `toolchain` package implies compilation happens, which is an important piece of information to make clear and distinct. By keeping the `toolchain` package for compiling, it makes it easy for someone to know if compilation is done in a recipe. We do not want to mix this up with Python packages and lose this very important piece of information. This package provided here is only useful for Python builds and does not configure compilers. Though the two could be combined together when both are needed.

The first piece added here are default `setuptools` arguments. One only needs to run `python setup.py install` and add `python-toolchain` as a build dependency to make this work. The only subtle point here is the configuration file we add seems to break `distutils`. However, I don't actually see this as a problem. We simply don't use this toolchain with `distutils`. Also, we can find out pretty quickly whether a package uses `setuptools` or not by doing this experiment. This saves us from hunting around in the code for this information. As we are going to disable the inclusion of `pip`, `setuptools`, and `wheel` as `python` dependencies on the VMs, it will be much easier to tell if something is building with `distutils` or not. Basically, people try not using `python-toolchain` (i.e. using `distutils`). When it fails, it will need the `python-toolchain` added too. The rest happens behind the scenes. We have simplified things for that use case by actually making `setuptools` a run dependency of the `python-toolchain` to ensure it gets pulled into so only the `python-toolchain` will be needed as a dependency. 

The second piece basically this implements the same idea that @minrk described in this [comment](https://github.com/conda-forge/staged-recipes/issues/528#issuecomment-216874083). This works by adding a configuration file to `$PREFIX/etc/pip.conf` or `%LIBRARY_PREFIX\etc\pip.conf` on Windows. As `pip` may not know where to find this file and it will not necessarily look in such places we set `PIP_CONFIG_FILE` in activation scripts to make sure it finds this. In the long run, we will want the functionality asked for in this issue ( https://github.com/conda/conda-build/issues/910 ) to make this work smoothly. In the interim, there is a kludgy one line hack that we can use to get by. In any event, this will allow one to do `pip install .` and everything will work correctly behind the scenes. Just as mentioned in the previous case, we made things easier by making `pip` a runtime dependency of the `python-toolchain`. Should add that regardless of whether the package uses `distutils` or `setuptools`, we still find `pip install .` works fine with the `python-toolchain`.

This package (just as the `toolchain` package) was designed so that it could work locally just as well as it does in the CIs. By doing this, we make it easy for people to debug things locally. A feature that remains useful for more complex problems.

Please provide feedback on this. Also, feel free to take it for a test drive. 🚗

cc @msarahan @ocefpaf @pelson @minrk @scopatz @frol

### CODE REVIEW

**Comment by jakirkham** [comment]
Running into this issue ( https://github.com/conda/conda-build/issues/963 ). Can come up with a workaround if needed.

**Comment by jakirkham** [comment]
Commented the failing tests for now.

**Comment by jakirkham** [comment]
Added some recipes for testing. Here `pint` uses `setuptools` and `webcolors` uses `distutils`.

**Comment by jakirkham** [comment]
Windows has some issue. I expect this is because I don't know how to activate it correctly. Maybe there are other issues too.

**Comment by pelson** [comment]
Excuse the extremely naive question, but does adopting `pip install . --no-deps` also solve the problem? I'm not hugely fond of hiding the configuration away from the recipe - I don't think it is a healthy approach to say `python setup.py install` just works, I can imagine consumers of recipes getting extremely confused.

**Comment by jakirkham** [comment]
> Excuse the extremely naive question, but does adopting `pip install . --no-deps` also solve the problem?

Unfortunately not. See this [comment]( https://github.com/conda-forge/staged-recipes/issues/528#issuecomment-216874083 ) by @minrk. Basically, to make sure things don't go sideways with `pip`, we need to add a bunch of options. The good news is that can be added to a config file, which is one thing we do here.

> I'm not hugely fond of hiding the configuration away from the recipe...

This really isn't magical. We are simply adding a [3 line global config file]( https://github.com/conda-forge/staged-recipes/blob/4aaaddb1e7301a168367123ddefb621ef8293da1/recipes/python-toolchain/distutils.cfg ). This is something that `setuptools` will always use. It lives in `$STDLIB_DIR/distutils/distutils.cfg`. So, it is not hidden. In fact this is the global configuration file.

> ...I don't think it is a healthy approach to say `python setup.py install` just works...

Unfortunately, adding these arguments ( `--single-version-externally-managed --record=record.txt` ) explicitly has gotten a bit divisive of late. I really don't want our community bikesheding these issues. One way to do that is configure processes to do this automatically. In this way, it is really no different from the `toolchain` package. Plus, I want to make sure it is easy to add recipes. The check-for-`distutils`-or-`setuptools`-in-the-code dance is getting a little tired. Combined with the fact that we are eliminating `setuptools` and `pip` as `python` dependencies, this becomes really simple. If you need `setuptools` you add `python-toolchain` to your build dependencies. If you don't, then you just don't install this. If you want to use `pip`, always add `python-toolchain` to your build dependencies.

> ...I can imagine consumers of recipes getting extremely confused.

Well, that's really a question of documentation. I don't see this as any more confusing than the `toolchain` at this point. Right now, I'm a bit more worried about the fact that Python recipes have no real standardization. This combined with eliminating `setuptools` and `pip` should help that.

<br>
In any event, I don't want us to get too hung up on this now as I would rather worry about the CI improvements that we are starting with `conda-forge-build-setup` first. That is a really big gain that we are going to need soon as we need to start using `conda-build` 1.20.3 and cleaning out some bottlenecks like the `curl` recipe.

File: recipes/python-toolchain/activate.bat [hunk 1]
```diff
@@ -0,0 +1 @@
+set PIP_CONFIG_FILE="%LIBRARY_PREFIX%\etc\pip.conf"
```
Comment (janschulz) [inline recipes/python-toolchain/activate.bat]: Usually it's `set "NAME=VALUE"`. No idea why, but using it that way prevented bugs in cmder

File: recipes/python-toolchain/bld.bat [hunk 2]
```diff
@@ -0,0 +1,16 @@
+setlocal EnableDelayedExpansion
+
+:: This works for `setuptools`, but breaks `distutils`.
+:: Will need to figure out a better long term strategy.
+copy "%RECIPE_DIR%\distutils.cfg" "%STDLIB_DIR%\distutils\distutils.cfg"
+
```
Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: if errorlevel 1 exit 1

File: recipes/python-toolchain/bld.bat [hunk 3]
```diff
@@ -0,0 +1,16 @@
+setlocal EnableDelayedExpansion
+
```
Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: This afaik only is useful when you use `!VAR!`instead of `%VAR%`. Also as far as I can see, there is no case below which would need delayed variables: in the for loop it's either a loop variable or a variable which was defined outside of the loop.

Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: I believe the `FOR` loop below failed without this so I had added it. Though I'm hardly a Windows expert so if you don't think it is needed we can try removing it.

File: recipes/python-toolchain/bld.bat [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+:: Configure `pip`.
+IF NOT EXIST "%LIBRARY_PREFIX%\etc" mkdir "%LIBRARY_PREFIX%\etc"
+copy "%RECIPE_DIR%\pip.conf" "%LIBRARY_PREFIX%\etc\pip.conf"
+
+:: Copy the [de]activate scripts to %LIBRARY_PREFIX%\etc\conda\[de]activate.d.
+:: This will allow them to be run on environment activation.
+FOR %%F IN (activate deactivate) DO (
+    IF NOT EXIST %LIBRARY_PREFIX%\etc\conda\%%F.d MKDIR %LIBRARY_PREFIX%\etc\conda\%%F.d
+    COPY %RECIPE_DIR%/%%F.bat %LIBRARY_PREFIX%\etc\conda\%%F.d\python-toolchain_%%F.bat
+)
```
Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: I  currently use this (without delayed expansion) in my go port (not yet pushed):

```bat
for %%F in (activate deactivate) do (
    if not exist "%LIBRARY_PREFIX%\etc\conda\%%F.d" mkdir "%LIBRARY_PREFIX%\etc\conda\%%F.d"
    copy "%RECIPE_DIR%\%%F.bat" "%LIBRARY_PREFIX%\etc\conda\%%F.d\go_%%F.bat"
)
if errorlevel 1 exit 1
```
-> different slashes in the copy line

Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: Good catch.

File: recipes/python-toolchain/activate.bat (hunk 1 above)
Comment (jakirkham) [inline recipes/python-toolchain/activate.bat]: Good point. I have seen that syntax for unsetting a variable, but not setting it. Will change.

File: recipes/python-toolchain/deactivate.bat [hunk 5]
```diff
@@ -0,0 +1 @@
+set "PIP_CONFIG_FILE="
```
Comment (janschulz) [inline recipes/python-toolchain/deactivate.bat]: Haven't seen quotes on unset (there are no spaces in this, so it should not be needed), but it will probably also not hurt :-)

**Comment by jakirkham** [comment]
Can you double check that these activation scripts work correctly on Windows, @janschulz? I took my best guess, but I'm still not totally sure that these work right or that I'm activating them correctly.

File: recipes/python-toolchain/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    - python
+    - setuptools
+    - pip
+
+test:
+  commands:
+    - test -f "${PREFIX}/etc/pip.conf"                                 # [unix]
+   #- test -f "${STDLIB_DIR}/distutils/distutils.cfg"                  # [unix]
+    - if not exist "%LIBRARY_PREFIX%\\etc\\pip.conf" exit 1            # [win]
+   #- if not exist "%STDLIB_DIR%\\distutils\\distutils.cfg" exit 1     # [win]
```
Comment (janschulz) [inline recipes/python-toolchain/meta.yaml]: In the go recipe I now try 
```
- cmd /c if x%GOROOT% NEQ x%CONDA_DEFAULT_ENV%\go exit 1   # [win]
```

Comment (janschulz) [inline recipes/python-toolchain/meta.yaml]: But it seems that we need a newer conda version for that to work: https://github.com/conda/conda/commit/f531272c24b8c123cab9ddd7c66154cba277d5cc

File: recipes/python-toolchain/bld.bat [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+:: Configure `pip`.
+IF NOT EXIST "%LIBRARY_PREFIX%\etc" mkdir "%LIBRARY_PREFIX%\etc"
+if errorlevel 1 exit 1
+copy "%RECIPE_DIR%\pip.conf" "%LIBRARY_PREFIX%\etc\pip.conf"
+if errorlevel 1 exit 1
+
+:: Copy the [de]activate scripts to %LIBRARY_PREFIX%\etc\conda\[de]activate.d.
+:: This will allow them to be run on environment activation.
+FOR %%F IN (activate deactivate) DO (
+    IF NOT EXIST "%LIBRARY_PREFIX%\etc\conda\%%F.d" MKDIR "%LIBRARY_PREFIX%\etc\conda\%%F.d"
```
Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: Oups, sorry, this should be `s/%LIBRARY_PREFIX%/%PREFIX%/` :-(

File: recipes/python-toolchain/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/python-toolchain/meta.yaml]: Yeah, I was doing activation manually in cases like that, which is horrendous, but it got the job done.

File: recipes/python-toolchain/bld.bat (hunk 7 above)
Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: Oh, really? That would explain some suffering that I am having with other activation scripts.

Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: At least the latest version of the [activation script](https://github.com/conda/conda/blob/master/bin/activate.bat#L51-L56) contains:

```
    @IF EXIST "%CONDA_NEW_ENV%\etc\conda\activate.d" (
```

`%CONDA_NEW_ENV% == %PREFIX%`

Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: Now with the fix in https://github.com/conda-forge/staged-recipes/pull/705#issuecomment-223464170 confirmed to work when the stuff is installed in `"%PREFIX%\etc\conda\%%F.d"`

File: recipes/python-toolchain/meta.yaml (hunk 6 above)
Comment (janschulz) [inline recipes/python-toolchain/meta.yaml]: Oh, and another problem: conda build does not activate the environment but only changes the PATH: at least the gopath isn't set in during tests :-(

```
The following NEW packages will be INSTALLED:

    go: 1.4.2-0

Fetching packages ...
go-1.4.2-0.tar 100% |###############################| Time: 0:00:00 130.22 MB/s
Extracting packages ...
[      COMPLETE      ]|##################################################| 100%
Linking packages ...
[      COMPLETE      ]|##################################################| 100%

C:\portabel\miniconda\conda-bld\test-tmp_dir>go version
go: cannot find GOROOT directory: C:\portabel\miniconda\envs\_build\go

C:\portabel\miniconda\conda-bld\test-tmp_dir>if errorlevel 1 exit 1
TESTS FAILED: go-1.4.2-0
```

I'm currently thinking about this workaround:

```
  commands:
    - activate _test  # [win]
    - go version
    - go help
    - go run hello.go
    - cmd /c echo x%GOROOT% NEQ x%CONDA_DEFAULT_ENV%\go  # [win]
    - cmd /c if x%GOROOT% NEQ x%CONDA_DEFAULT_ENV%\go exit 1   # [win]
```

Comment (janschulz) [inline recipes/python-toolchain/meta.yaml]: And the workaround works :-)

File: recipes/python-toolchain/bld.bat (hunk 7 above)
Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: So, we are nearing a `conda` 4.1 release so we may see this soon.

File: recipes/python-toolchain/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/python-toolchain/meta.yaml]: Sorry, I confused the two issues. Also, using `activate` to solve this problem on *NIXes. This is the `conda-build` issue ( https://github.com/conda/conda-build/issues/910 ) that pertains to fixing this problem.

File: recipes/python-toolchain/bld.bat [hunk 8]
```diff
@@ -0,0 +1,19 @@
+:: This works for `setuptools`, but breaks `distutils`.
+:: Will need to figure out a better long term strategy.
+copy "%RECIPE_DIR%\distutils.cfg" "%STDLIB_DIR%\distutils\distutils.cfg"
+if errorlevel 1 exit 1
+
+:: Configure `pip`.
+IF NOT EXIST "%LIBRARY_PREFIX%\etc" mkdir "%LIBRARY_PREFIX%\etc"
+if errorlevel 1 exit 1
+copy "%RECIPE_DIR%\pip.conf" "%LIBRARY_PREFIX%\etc\pip.conf"
+if errorlevel 1 exit 1
```
Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: Should these also be `%PREFIX%`, @janschulz? We set an environment variable to tell it where to look, but I want to make sure we are doing a sensible thing here. Please let me know your thoughts.

File: recipes/python-toolchain/bld.bat (hunk 7 above)
Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: Awesome. Thanks for your guidance here, @janschulz. This has been updated respectively. Please help me double check that this is now correct in the new diff.

**Comment by jakirkham** [comment]
Seems that we are still suffering form activation issues on Windows. Suggestions?

File: recipes/python-toolchain/bld.bat (hunk 8 above)
Comment (janschulz) [inline recipes/python-toolchain/bld.bat]: no idea:

The only packages I have here which install etc directories are 

`m2-filesystem-2016.04-3\Library\etc\` -> all kind of bash stuff
`go-1.4.2-0\etc\conda\` -> The activate.d stuff...
 

I also have
* `<env>\share` -> jupyter stuff, like `...\share\jupyter\kernels\python3` and some man page in `..\share\man\man1\...`
* `<env>\Skripts` for the binaries which should become available in PATH

and

* `<env>\Library\bin` (which is also in PATH), but seems to only have qt stuff (designer, etc) and some dlls in it
* `<env>\Library\share\man\` -> libpng man pages
* `<env>\Library\share\pkgconfig`

So basically it seems to be "undefined" :-)

**Comment by janschulz** [comment]
> Seems that we are still suffering form activation issues on Windows. Suggestions?

```
commands:
    - activate _test  # [win]
    - python tests.py # ot whatever...
```

And something similar in `bld.bat`?  

That's a horrible workaround, but it seems to work :-)

**Comment by jakirkham** [comment]
@msarahan, do you see anything wrong with the Windows activation here? I think the paths are all correct, but I could still be missing something. If you can see something that is obviously wrong, please let me know. As this would be something using PR ( https://github.com/conda/conda-build/pull/1002 ), it would be good if we can make sure this is working correctly. Maybe we are running into common issues.

File: recipes/webcolors/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+source:
+  fn: webcolors-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/w/webcolors/webcolors-{{ version }}.tar.gz
+  md5: 2e9da1bbb7a087c6108e05be64640afb
+
+build:
+  number: 0
+  script:
+    - source activate _build  # [unix]
+    - activate _build         # [win]
```
Comment (msarahan) [inline recipes/webcolors/meta.yaml]: Nothing obvious.  What happens if you just do

activate _build 

Interactively?

Comment (jakirkham) [inline recipes/webcolors/meta.yaml]: Good question. If I had a working Windows machine, I could tell you. 😉 Unfortunately, no such luck. 😞

Comment (jakirkham) [inline recipes/webcolors/meta.yaml]: Though I do have a sort of working Windows machine (think the compiler installs didn't go over well 😨). Fortunately, this doesn't require working compilers (I would hope 👼). FWICT it is not activating correctly there. 😕

Comment (msarahan) [inline recipes/webcolors/meta.yaml]: Don't forget you can log into appveyor directly:
https://www.appveyor.com/docs/how-to/rdp-to-build-worker

Comment (janschulz) [inline recipes/webcolors/meta.yaml]: I think I remember fixing my activate script as the currently released version of conda/conda-env does not have the activate.d step as part of the `endlocal` line (line = `endlocal && (....)` and therefore the nothing which happens in the activate.d scripts is surviving until the activate script returns.

https://github.com/conda/conda/commit/f531272c24b8c123cab9ddd7c66154cba277d5cc

-> 
bad
```
endlocal & (
    REM Used for deactivate, to make sure we restore original state after deactivation
    SET "CONDA_PATH_BACKUP=%CONDA_PATH_BACKUP%"
    SET "CONDA_OLD_PS1=%CONDA_OLD_PS1%"
    set "PROMPT=%PROMPT%"
    set "PATH=%PATH%"
    set "CONDA_DEFAULT_ENV=%CONDA_NEW_ENV%"
    )
```

workaround:
```
endlocal & (
    REM Used for deactivate, to make sure we restore original state after deactivation
    SET "CONDA_PATH_BACKUP=%CONDA_PATH_BACKUP%"
    SET "CONDA_OLD_PS1=%CONDA_OLD_PS1%"
    set "PROMPT=%PROMPT%"
    set "PATH=%PATH%"
    set "CONDA_DEFAULT_ENV=%CONDA_NEW_ENV%"
    )

REM Run any activate scripts outside of "setlocal"
IF EXIST "%CONDA_DEFAULT_ENV%\etc\conda\activate.d" (
    PUSHD "%CONDA_DEFAULT_ENV%\etc\conda\activate.d"
    FOR %%g in (*.bat) DO @CALL "%%g"
    POPD
)

```

**Comment by janschulz** [comment]
I think I remember fixing my activate script as the currently released version of conda/conda-env does not have the activate.d step as part of the `endlocal` line (line = `endlocal && (....)` even if there are line breaks) and therefore the nothing which happens in the activate.d scripts is surviving until the activate script returns.

https://github.com/conda/conda/commit/f531272c24b8c123cab9ddd7c66154cba277d5cc

-> workaround:
```
endlocal & (
    REM Used for deactivate, to make sure we restore original state after deactivation
    SET "CONDA_PATH_BACKUP=%CONDA_PATH_BACKUP%"
    SET "CONDA_OLD_PS1=%CONDA_OLD_PS1%"
    set "PROMPT=%PROMPT%"
    set "PATH=%PATH%"
    set "CONDA_DEFAULT_ENV=%CONDA_NEW_ENV%"
    )

REM Run any activate scripts outside of "setlocal"
IF EXIST "%CONDA_DEFAULT_ENV%\etc\conda\activate.d" (
    PUSHD "%CONDA_DEFAULT_ENV%\etc\conda\activate.d"
    FOR %%g in (*.bat) DO @CALL "%%g"
    POPD
)

```

**Comment by jakirkham** [comment]
Yeah, I was about to say I think this is why I had `setlocal EnableDelayedExpansion` added. I'm going to try bringing that back.

**Comment by jakirkham** [comment]
Yeah, otherwise we see these really weird [`if not exist` statements]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.3017/job/e3qaqy8vdvr05y1a#L495 ) that weren't part of the original script.

**Comment by jakirkham** [comment]
Nope, I'm wrong that makes no difference. Same problem.

**Comment by janschulz** [comment]
as far as I understand batch, the only think to make it work would be to hack the activate.bat script.

**Comment by janschulz** [comment]
Re the `if not exist` parts: aren't that the tests? `if not exist "%LIBRARY_PREFIX%\\etc\\pip.conf" exit 1`

File: recipes/python-toolchain/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  run:
+    - python
+    - setuptools
+    - pip
+
+test:
+  commands:
+    - test -f "${PREFIX}/etc/pip.conf"                                 # [unix]
+   #- test -f "${STDLIB_DIR}/distutils/distutils.cfg"                  # [unix]
+    - if not exist "%LIBRARY_PREFIX%\\etc\\pip.conf" exit 1            # [win]
```
Comment (janschulz) [inline recipes/python-toolchain/meta.yaml]: According to the appveyor log you don't need `\\`, `\` is enough.

**Comment by jakirkham** [comment]
So, basically, we need a `conda` release before we can do anything with Windows really. IMHO we should not be asking people to add such a long chunk of stuff in `bld.bat` to effectively activate and deactivate things.

File: recipes/python-toolchain/meta.yaml (hunk 10 above)
Comment (jakirkham) [inline recipes/python-toolchain/meta.yaml]: I'm a little more worried about something parsing the yaml and misinterpreting something. As AppVeyor can handle this fine, I'm happy enough to leave this alone.

**Comment by janschulz** [comment]
So, IMO currently (with the current `conda build` and `conda/conda env`) there are two issues which make this fail:

* `conda build` does not activate the build environments, only sets the PATH and some other stuff, but doesn't get anything else from `activate.d` scripts (https://github.com/conda/conda-build/issues/910)
* but even if it would do this (or the tests use the workaround of activating the env actively), the `activate.d` scripts can't influence the environment as they are currently run inside the `setlocal` scope. (fixed in master, but apparently not yet released?)

**Comment by jakirkham** [comment]
You're right. I guess the copying is buried in these [lines]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.3019/job/95hcljgf8si65xfg#L467 ). Maybe I should enable some sort of verbose flag so it is easier to read.

**Comment by janschulz** [comment]
Thats actually I also noticed: the bld.bat script is not anymore run with `echo on`?

**Comment by jakirkham** [comment]
I don't think I ever had `echo on`. Trying to add it now.

**Comment by jakirkham** [comment]
Anyways `conda` 4.1 and `conda-build` 2.0.0 are on there way. We know half the problem is solved already. We just need to solve the other half.

File: recipes/python-toolchain/bld.bat (hunk 8 above)
Comment (jakirkham) [inline recipes/python-toolchain/bld.bat]: @msarahan, any thoughts on where `pip.conf` should live? Normally this is in something like `$PREFIX/etc`. Though I don't know whether we should use `%PREFIX%\etc` or `%LIBRARY_PREFIX%\etc` on Windows.

**Comment by jakirkham** [comment]
It might be time to pick this up again as the needed changes for Windows are in `conda`/`conda-build`.

**Comment by jakirkham** [comment]
Rebased on [current `master`]( https://github.com/conda-forge/staged-recipes/commit/b075603c59b01f8e0d70660445831cfa933aaa3f ). Though we may still have some issues with AppVeyor and activation scripts.
